### PR TITLE
Remove the old DEPRECATEDIN macros

### PR DESCRIPTION
**Pull request by levitte**
They serve no purpose any more

### CODE REVIEW

**Comment by levitte** [comment]
Now that all uses of the old macros have been refactored to the new form, we can finally remove the old.
I just rebased this PR, and the CIs should confirm that this is indeed fine at this point.

**Comment by richsalz** [comment]
And these?
```
util/mkdef.pl:            next unless /^DEPRECATEDIN_(\d+)_(\d+)(?:_(\d+))?$/;
util/perl/OpenSSL/ParseC.pm:    # DEPRECATEDIN_x_y[_z].
util/perl/OpenSSL/ParseC.pm:    { regexp   => qr/(DEPRECATEDIN_\d+_\d+(?:_\d+)?)<<<\((.*)\)>>>/,
```

**Comment by levitte** [comment]
Well spotted.  Will fix promptly

**Comment by levitte** [comment]
The mkdef.pl one isn't a C macro name, it's a feature name in util/*.num, so it doesn't count.

**Comment by levitte** [comment]
Done

File: apps/lib/http_server.c [hunk 1]
```diff
@@ -54,7 +54,7 @@ void log_message(const char *prog, int level, const char *fmt, ...)
     if (multi) {
         char buf[1024];
 
-        if (vsnprintf(buf, sizeof(buf), fmt, ap) > 0)
+        if (BIO_vsnprintf(buf, sizeof(buf), fmt, ap) > 0)
```
Comment (t8m) [inline apps/lib/http_server.c:57]: This is completely unrelated so it should be a separate commit, shouldn't it?

File: apps/lib/names.c [hunk 2]
```diff
@@ -8,6 +8,7 @@
  */
 
 #include <string.h>
+#include <strings.h>             /* strcasecmp() */
```
Comment (t8m) [inline apps/lib/names.c:11]: Similarly here and in other files where strings.h is added or `#define _XOPEN_SOURCE 500` is added.

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by levitte** [comment]
@t8m, you were right, those mysterious additions was a mistake.  Now undone.

**Comment by levitte** [comment]
Ping

**Review by t8m — APPROVED** [review]

**Comment by levitte** [comment]
Merged

1e3affbbcd85856c78e50c6bf56144bf9bc0eb23 Remove the old DEPRECATEDIN macros

**Pull request closed without merging**